src/CMakeLists.txt

```diff
@@ -0,0 +1,183 @@
+    rad/designer/resizablepanel.cpp
+    rad/designer/visualeditor.cpp
+    rad/designer/innerframe.cpp
+    rad/designer/visualobj.cpp
+    rad/designer/menubar.cpp
+    rad/luapanel/luapanel.cpp
+    utils/filetocarray.cpp
+    utils/annoyingdialog.cpp
+    utils/stringutils.cpp
+    utils/encodingutils.cpp
+    utils/wxfbipc.cpp
+    utils/typeconv.cpp
+    codegen/phpcg.cpp
+    codegen/codeparser.cpp
+    codegen/xrccg.cpp
+    codegen/luacg.cpp
+    codegen/cppcg.cpp
+    codegen/codegen.cpp
+    codegen/pythoncg.cpp
+    codegen/codewriter.cpp
+    model/xrcfilter.cpp
+    model/objectbase.cpp
+    model/types.cpp
+    model/database.cpp
+)
+
+if(WIN32)
+    list(APPEND wxFB_SRCS wxFormBuilder.rc)
+elseif(APPLE)
+    set(wxFB_SRCS ${wxFB_SRCS} wxFormBuilder.icns)
+    set(MACOSX_BUNDLE_ICON_FILE wxFormBuilder.icns)
+    set_source_files_properties(wxFormBuilder.icns PROPERTIES
+        MACOSX_PACKAGE_LOCATION Resources)
+
+    file(GLOB XMLS ../output/xml/*)
+    list(APPEND wxFB_SRCS ${XMLS})
+    set_source_files_properties(${XMLS} PROPERTIES MACOSX_PACKAGE_LOCATION "SharedSupport/xml")
+    source_group("Resources\\xml" FILES ${XMLS})
+
+    file(GLOB RES LIST_DIRECTORIES false ../resources/*)
+    list(APPEND wxFB_SRCS ${RES})
+    set_source_files_properties(${RES} PROPERTIES MACOSX_PACKAGE_LOCATION "SharedSupport/resources")
+    source_group("Resources" FILES ${RES})
+
+    file(GLOB ICONS LIST_DIRECTORIES false ../resources/icons/*)
+    list(APPEND wxFB_SRCS ${ICONS})
+    set_source_files_properties(${ICONS} PROPERTIES MACOSX_PACKAGE_LOCATION "SharedSupport/resources/icons")
+    source_group("Resources\\Icons" FILES ${ICONS})
+endif()
+
+add_executable(wxFormBuilder WIN32 MACOSX_BUNDLE
+    ${wxFB_SRCS}
+)
+
+if(APPLE)
+    set_target_properties(wxFormBuilder PROPERTIES
+        MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.in"
+        MACOSX_BUNDLE_GUI_IDENTIFIER "org.wxformbuilder.rad"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "${CPACK_PACKAGE_VERSION}"
+        )
+endif()
+
+target_include_directories(wxFormBuilder
+    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+target_link_libraries(wxFormBuilder
+    ${wxWidgets_LIBRARIES}
+    ticpp
+    wxfb-plugin-interface
+)
+
+if(MINGW)
+    find_path(BINUTILS_BASE_INCLUDE_DIR binutils/bfd.h)
+    target_include_directories(wxFormBuilder
+        PRIVATE ${BINUTILS_BASE_INCLUDE_DIR}/binutils
+    )
+endif()
+
+if(WIN32)
+    if(MSVC)
+        target_link_libraries(wxFormBuilder Dbghelp)
+    else()
+        find_library(BFD_LIB NAMES binutils/libbfd.a libbfd.a)
+        find_library(IBERTY_LIB NAMES binutils/libiberty.a libiberty.a)
+        find_package(ZLIB REQUIRED)
+        find_package(Intl REQUIRED)
+
+        target_link_libraries(wxFormBuilder
+            ${BFD_LIB}
+            ${IBERTY_LIB}
+            ${Intl_LIBRARIES}
+            ${ZLIB_LIBRARIES}
+            psapi
+            imagehlp)
+    endif()
+endif()
+
+if(CMAKE_GENERATOR MATCHES "^MSYS")
```

Comment:
Hm ... what if someone uses Ninja on MSYS2?

Reply:
i don't use msys, so no clue.

its a regex, so it should match MSYS and MSYS2